### ISSUE TITLE
 failure when "copy image" to word

### ISSUE THREAD

**Issue by gd2019xyz** [rstudio/rstudio#5103]
### System details

    RStudio Edition : 
    RStudio Version : 1.2.1335
    OS Version      :  3.6.1
    R Version       :  Windows 10 x64 (build 17134)

I create a plot and try to copy it to a word. but the plot can not display
what I did is :

1, click zoom , right click and select "copy image"
![capture1](https://user-images.githubusercontent.com/52926043/61227537-1b2cf500-a6ea-11e9-8903-ea86cce3966d.PNG)

2, paste it in a word, but it can not display. my word is office 2010
![capture2](https://user-images.githubusercontent.com/52926043/61227686-634c1780-a6ea-11e9-8a3d-0d999e22a513.PNG)

**Comment by kevinushey** [comment]
Are you able to successfully paste the image into other applications? I don't have Microsoft Office on my Windows VM but I was able to successfully paste the image into e.g. Paint.

**Comment by gd2019xyz** [comment]
@kevinushey  thank your reply.
I try paste  the image into paint,  and then copy it into word.  it work!!!.  but it is a little  inconvenient.  
by the way, when I "save image as" into disk. I find the saved file has not extensions. I guess that is why it is failed when I "copy image" to word directly

**Comment by ronblum** [comment]
@gd2019xyz and @kevinushey I can reproduce this when copying from RStudio Desktop 1.2.1558 into Word and Excel 16.26 on MacOS 10.15 (Catalina). I'm able to paste this into a Gmail email in Google Chrome.

Also, note that the "Save Image As" called the file `plot_zoom_png` by default, with no file extension. It should probably be `plot_zoom.png`. But that's a separate issue.

**Comment by Santihago** [comment]
Same problem with RStudio Desktop (Version 1.2.1335) on Mac OS X 10.14.5 when I try to copy an image directly to Word (Version 16.24). Copying elsewhere (**Preview** on OS X) works.

**Comment by jmcphers** [comment]
Possibly related (and/or duplicate): https://github.com/rstudio/rstudio/issues/5712

**Comment by AndrewTheTM** [comment]
I have the same problem, but I can get the plot image to paste by using Paste Special - Image in Word. This seems to have started since the version that was out in November 2018 (I wish I knew which version - I didn't install it; I just upgraded to 1.2.5033 and this started on me with this version).

**Comment by ndevln** [comment]
I have the same problem on 2 Windows 10 Machines at least since the first stable RStudio 1.2 release. Pasting into LibreOffice Writer still works. You can even paste it into Writer, then copy it again and paste it into Word. (A workaround for the desperate.)

**Comment by GBeattie** [comment]
I found the same problem when trying to copy over images from the Rstudio Zoom window to PowerPoint. I've found two simple workarounds: Paste Special and choose TIFF, or (my preference) drag the image from the zoom window and drop in PP. Hopefully the same approaches work in Word.

**Comment by Stevepelcz** [comment]
In Word, selecting "Home > Paste Special > Device Independent Bitmap" was the solution for me.

**Comment by timtmok** [comment]
I think this is a result of how Chromium handles the copy image. We might be able to do something with Electron to override the action and update the clipboard using Electron API. I think the workaround is a suitable solution for now and others have been using special paste when copying from Chrome to Word.

I do wonder how well this would work with Edge now that it is Chromium based (the image to copy should be PNG for an equivalent test). It's likely that Word doesn't recognize the clipboard contents to understand the image format when using regular paste.

**Comment by ronblum** [comment]
The issue is still present in 
- RStudio Desktop 2022.11.0+98 on MacOS 12.5
- RStudio Desktop 2022.11.0+101 on Windows 11.

**Comment by kevinushey** [comment]
This is where "Copy Image" is handled:

https://github.com/rstudio/rstudio/blob/aefa76732fd47f130cd85bda44364a020923b9d3/src/node/desktop/src/main/context-menu.ts#L100-L106

I wonder if we can do something like:

1. Send focus to the location where the context menu was generating (simulating a left mouse click),
2. Execute the default clipboard "copy" action.

Maybe this will run through a different Electron code path that will copy the image in a better form?

**Comment by smouksassi** [comment]
2023.03.0 Build 386 still has the same issue

**Comment by olivroy** [comment]
I think this would be great if this was fixed.
I tried 4 ways to copy an image to Word and Outlook

1-2-3 seem to be routed through the same code as I see no difference in behavior. 4-5  are just (inefficient) workarounds
1. Copy image (right click)
2. Drag (same thing happens)
3. Export -> Copy to Clipboard  (same thing happens)
4. Export -> Save as Image -> select a folder outside my project's directory not to pollute my project + select open file. Hopefully that now works (#14702) 
5. Copy in another app -> recopy to my email or word document.

**Comment by kevinushey** [comment]
It seems to work for me (pasting into Word). I have to select the "Picture" paste option when pasting:

![image](https://github.com/rstudio/rstudio/assets/1976582/0dfdc534-e744-42d7-8ace-392fc2732a5b)
 
since it seems that (speculating) Word wants to use the text clipboard, even if there is no text on the clipboard?

**Comment by olivroy** [comment]
Indeed, it works this way.. I did some more testing and apparently, it is just the web-based Outlook (both M365 and outlook.live.com) that doesn't work correctly. (Their paste as image doesn't work) 

Pasting in Teams chat works out of the box as other online M365 apps. Sorry for the wrong diagnostic.
